### PR TITLE
Allow Pin to switch the Nearest Pin or Hot Pin

### PR DESCRIPTION
**Pull request by WeiJun0507**

### CODE REVIEW

File: lib/pages/home/pins.dart [hunk 1]
```diff
@@ -19,12 +19,88 @@ class PinsPage extends StatefulWidget {
...
 class _PinsPageState extends State<PinsPage> {
-  late final LoadingBase<PinItemModel> _lb = LoadingBase(
+  LoadingBase<PinItemModel> _lb = LoadingBase(
     request: (_, String? lastId) => RecommendAPI.getRecommendPins(
       lastId: lastId,
+      sortType: 200,
     ),
   );
 
+  set lb(LoadingBase<PinItemModel> value) {
```
Comment (shirne) [inline lib/pages/home/pins.dart]: It's unnessly to reset `_lb`.
Just set a variable `sortType` would be ok.

File: lib/apis/recommend_api.dart [hunk 2]
```diff
@@ -7,6 +7,19 @@ import '../models/data_model.dart';
 import '../models/response_model.dart';
 import '../utils/http_util.dart';
 
+enum SortType {
+  hot_pin(200),
```
Comment (AlexV525) [inline lib/apis/recommend_api.dart]: Use `camelCase` as much as possible.

File: lib/apis/recommend_api.dart [hunk 3]
```diff
@@ -7,6 +7,19 @@ import '../models/data_model.dart';
...
+  newest_pin(300);
+
+  const SortType(this.value);
+  final int value;
```
Comment (AlexV525) [inline lib/apis/recommend_api.dart:16]: Need a blank line before this.

File: lib/pages/home/pins.dart [hunk 4]
```diff
@@ -19,29 +19,28 @@ class PinsPage extends StatefulWidget {
...
       width: 80,
       padding: const EdgeInsets.symmetric(horizontal: 4.0),
       decoration: BoxDecoration(
           borderRadius: RadiusConstants.r20,
-          color: Colors.grey[200]
+          color: Colors.grey[200],
       ),
       child: Stack(
-        alignment: _hotPin ? AlignmentDirectional.centerEnd : AlignmentDirectional.centerStart,
+        alignment: _sortType == SortType.hot_pin ? AlignmentDirectional.centerEnd : AlignmentDirectional.centerStart,
```
Comment (AlexV525) [inline lib/pages/home/pins.dart]: Why should we switch Stack's alignment?

File: lib/pages/home/pins.dart [hunk 5]
```diff
@@ -82,7 +72,7 @@ class _PinsPageState extends State<PinsPage> {
             ),
           ),
           Container(
-            alignment: _hotPin ? Alignment.centerRight : Alignment.centerLeft,
+            alignment: _sortType == SortType.hot_pin ? Alignment.centerRight : Alignment.centerLeft,
```
Comment (AlexV525) [inline lib/pages/home/pins.dart]: Use Directional as much as possible.

File: lib/apis/recommend_api.dart [hunk 6]
```diff
@@ -7,6 +7,19 @@ import '../models/data_model.dart';
...
+
+  @override
+  String toString() => value.toString();
+
```
Comment (AlexV525) [inline lib/apis/recommend_api.dart]: Format codes before committing.

File: lib/apis/recommend_api.dart [hunk 7]
```diff
@@ -7,6 +7,18 @@ import '../models/data_model.dart';
 import '../models/response_model.dart';
 import '../utils/http_util.dart';
 
+enum SortType {
+  hotPin(200),
+  newestPin(300);
```
Comment (shirne) [inline lib/apis/recommend_api.dart]: The enum value is used for all content type. Should not with `Pin` in name

**Review by shirne — APPROVED** [review]

**Review by shirne — APPROVED** [review]

File: lib/apis/recommend_api.dart [hunk 8]
```diff
@@ -7,6 +7,19 @@ import '../models/data_model.dart';
...
+  topic(200),
+  hot(200),
+  newest(300);
+
+  const SortType(this.value);
+
+  final int value;
+
+  @override
+  String toString() => value.toString();
```
Comment (AlexV525) [inline lib/apis/recommend_api.dart]: What's the overridden `toString()` for?

File: lib/apis/recommend_api.dart [hunk 9]
```diff
@@ -7,6 +7,19 @@ import '../models/data_model.dart';
 import '../models/response_model.dart';
 import '../utils/http_util.dart';
 
+enum SortType {
+  topic(200),
+  hot(200),
```
Comment (AlexV525) [inline lib/apis/recommend_api.dart]: Why do we have the same value for two instances?

File: lib/apis/recommend_api.dart [hunk 10]
```diff
@@ -7,6 +7,19 @@ import '../models/data_model.dart';
...
+  newest(300);
```
Comment (AlexV525) [inline lib/apis/recommend_api.dart]: Should be `latest`

File: lib/pages/home/pins.dart [hunk 11]
```diff
@@ -19,12 +19,77 @@ class PinsPage extends StatefulWidget {
...
+      sortType: _sortType,
     ),
   );
 
+  Widget _buildPinSelectedSwitch(BuildContext context) {
+    return Container(
+      width: 80,
+      padding: const EdgeInsets.symmetric(horizontal: 4.0),
+      decoration: BoxDecoration(
+          borderRadius: RadiusConstants.r20,
```
Comment (AlexV525) [inline lib/pages/home/pins.dart]: These are not well-aligned.

File: lib/pages/home/pins.dart [hunk 12]
```diff
@@ -34,6 +99,27 @@ class _PinsPageState extends State<PinsPage> {
...
+              borderRadius: RadiusConstants.r10,
+              color: context.theme.cardColor,
+            ),
+            child: Row(
+              children: <Widget>[
+                Expanded(
+                  child: Text(
+                    '排序',
+                    style: context.textTheme.caption?.copyWith(
+                      color: Colors.black,
```
Comment (AlexV525) [inline lib/pages/home/pins.dart]: Doesn't seem to be a valid color.

**Review by AlexV525 — CHANGES_REQUESTED** [review]

File: lib/apis/recommend_api.dart (hunk 9 above)
Comment (shirne) [reply lib/apis/recommend_api.dart]: for article they named `推荐`
for pin they named `热门`

Comment (WeiJun0507) [reply lib/apis/recommend_api.dart]: topic and hot serve for different pages but with the same value

Comment (AlexV525) [reply lib/apis/recommend_api.dart]: Use `recommend` then, same value will cause more confuse.

**Review by shirne — APPROVED** [review]

File: lib/l10n/app_en.arb [hunk 13]
```diff
@@ -64,6 +64,10 @@
   "userNotFollow": "Not follow",
   "userSignInOrUp": "Sign in/Sign up",
 
+  "pinArrangement": "Sort By",
+  "pinLatest": "Hot",
```
Comment (shirne) [inline lib/l10n/app_en.arb]: Whats different with latest and newest ?
May be recommend or hot?

File: lib/l10n/app_zh.arb [hunk 14]
```diff
@@ -53,6 +53,11 @@
   "userNotFollow": "未关注",
   "userSignInOrUp": "登录/注册",
 
+  "pinArrangement": "排序",
+  "pinLatest": "热门",
+  "pinNewest": "最新",
```
Comment (AlexV525) [inline lib/l10n/app_zh.arb]: These name can get rid of "pin".

File: lib/pages/home/pins.dart [hunk 15]
```diff
@@ -19,12 +19,107 @@ class PinsPage extends StatefulWidget {
 }
 
 class _PinsPageState extends State<PinsPage> {
+  SortType _sortType = SortType.latest;
+
+  set sortType(SortType value) {
+    _sortType = value;
+  }
```
Comment (AlexV525) [inline lib/pages/home/pins.dart]: Redundant settter.

File: lib/pages/home/pins.dart [hunk 16]
```diff
@@ -19,12 +19,107 @@ class PinsPage extends StatefulWidget {
...
+        alignment: _sortType == SortType.recommend
+            ? AlignmentDirectional.centerEnd
+            : AlignmentDirectional.centerStart,
+        children: [
+          Container(
+            margin: const EdgeInsets.all(2.0),
+            padding: const EdgeInsets.symmetric(
+              horizontal: 4.0,
+              vertical: 4.0,
+            ),
```
Comment (AlexV525) [inline lib/pages/home/pins.dart]:
```suggestion
            padding: const EdgeInsets.all(4),
```

File: lib/pages/home/pins.dart [hunk 17]
```diff
@@ -19,12 +19,107 @@ class PinsPage extends StatefulWidget {
...
     ),
   );
 
+  Widget _buildPinSelectedSwitch(BuildContext context) {
+    return Container(
+      width: 90,
+      padding: const EdgeInsets.symmetric(horizontal: 4.0),
+      decoration: BoxDecoration(
+        borderRadius: RadiusConstants.r20,
+        color: Colors.grey[200],
```
Comment (AlexV525) [inline lib/pages/home/pins.dart]: Could we use color from the theme?

File: lib/pages/home/pins.dart [hunk 18]
```diff
@@ -19,12 +19,107 @@ class PinsPage extends StatefulWidget {
...
+          Container(
+            width: _sortType == SortType.recommend ? 35 : 45,
+            height: 20,
+            decoration: const BoxDecoration(
+              borderRadius: RadiusConstants.r10,
+              color: Colors.white,
+            ),
+            child: Center(
+              child: Text(
+                _sortType == SortType.recommend ? context.l10n.pinLatest : context.l10n.pinNewest,
```
Comment (AlexV525) [inline lib/pages/home/pins.dart]: The line seems to be overflowed with 80 indent.

**Review by AlexV525 — COMMENTED** [review]
Please provide screenshots after the new commits with the new interface the PR implemented.

File: lib/pages/home/pins.dart (hunk 18 above)
Comment (WeiJun0507) [reply lib/pages/home/pins.dart]: What does the meaning of overflow with 80 indent? I used Reformat code using 'dart format' and it refactors the code in this way

Comment (AlexV525) [reply lib/pages/home/pins.dart]: Means your indent configuration is not 80 for dart files.

**Comment by WeiJun0507** [comment]
https://user-images.githubusercontent.com/66726409/188794326-284cb727-31f8-494c-94e1-5ea4b9ca9b95.mp4

This is the effect of the Pin Switching Tab

**Comment by shirne** [comment]
@AlexV525  here is screen record from @WeiJun0507


https://user-images.githubusercontent.com/2263157/188795812-92097e1d-07f1-44e2-a920-2d71f99276da.mp4

**Review by AlexV525 — APPROVED** [review]

**Comment by AlexV525** [comment]
@all-contributors Add @WeiJun0507 for code.

**Pull request merged**